static/js/main.js

```diff
@@ -128,43 +131,86 @@ function goBack() {
     var page = document.getElementById('mainb');
     page.innerHTML = `<h1 id='tabTitle'>Xbee Nodes</h1>
     <div id='cTable'></div>
-    <button onclick='refresh()'>Refresh</button>`;
+    <p id='refbutton'><button onclick='refresh()'>Refresh</button></p>`;
     var table = document.getElementById('cTable');
     table.innerHTML = tabletext;
 }
 
-function getNodes() {
+function getNodes(refreshtest){
+    //refreshtest just determines what function to send data to.
+    //true for refresh(), false for initialize().
     var jsonobj = '';
     var xhttp = new XMLHttpRequest();
 
-    xhttp.onreadystatechange = function() {
-        if (xhttp.readyState == XMLHttpRequest.DONE && xhttp.status == 200) {
+    xhttp.onreadystatechange = function(){
+        if(xhttp.readyState == XMLHttpRequest.DONE && xhttp.status == 200){
             jsonobj = JSON.parse(xhttp.responseText);
-            console.log(jsonobj);
+            if(refreshtest && jsonobj.success){
+                newxBeeArray = jsonobj.nodes;
+                refresh();
+            }else if(!refreshtest && jsonobj.success){
+                xBeeArray = jsonobj.nodes;
+                initialize();
+            }else{
+                alert('You are not authorized, please sign in.');
+                window.location = 'login.html';
+            }
         }
     };
     xhttp.open('GET', '/api/list', true);
     xhttp.send();
     return jsonobj;
 }
 
-function login(loginfo) {
+function login(){
     var jsonobj;
+    var loginfo = {"user" : $('#user').val(), "pass" : $('#password').val()};
     var xhttp = new XMLHttpRequest();
-    xhttp.onreadystatechange = function() {
-        if (xhttp.readyState == XMLHttpRequest.DONE && xhttp.status == 200) {
+    xhttp.onreadystatechange = function(){
+        if(xhttp.readyState == XMLHttpRequest.DONE && xhttp.status == 200){
             jsonobj = JSON.parse(xhttp.responseText);
-            console.log(jsonobj);
+            if(jsonobj.success){
+              window.location = 'index.html';
+            }else{
+              $( '#errorcon' ).text( 'Error: ' + jsonobj.error );
+            }
         }
     };
-    xhttp.open('POST', '/api/login', true);
-    xhttp.send(loginfo);
-    return jsonobj;
+    xhttp.open('POST','/api/login', true);
+    xhttp.setRequestHeader("Content-Type", "application/json");
+    xhttp.send(JSON.stringify(loginfo));
+}
+
+function logout(){
+    var jsonobj = '';
+    var xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange = function(){
+        if(xhttp.readyState == XMLHttpRequest.DONE && xhttp.status == 200){
+            jsonobj = JSON.parse(xhttp.responseText);
+            if(jsonobj.success){
+                window.location = 'login.html';
+            }
+        }
+    };
+    xhttp.open('GET','/api/logout', true);
+    xhttp.send();
 }
 
+function addXbees(i){
+    jsonobj = '';
+    var xhttp = new XMLHttpRequest();
+
+    xbee = {"node_id":i, "name":i.toString(), "units":i.toString()};
```

Comment:
Same, but `xbee`. Also not sure why the test data uses the same information for all fields.